ui/components/TracePage/TracePage.tsx

```diff
@@ -207,6 +283,10 @@ const useTreeContext = () => {
   return context;
 };
 
+function useBlockValue<T>(blockAddress: BlockAddress<T>): T | undefined {
```

Comment:
Some of the uses of this hook violate the [rules of hooks](https://reactjs.org/docs/hooks-rules.html). I think it might need to be restructured so a single hook invocation (at the top level) can handle a variable number of block addresses. Want to chat about this?